cli/bpmetadata/types.go

```diff
@@ -4,92 +4,141 @@ import (
-	Diagrams      []BlueprintDiagram     `json:",omitempty" yaml:",omitempty"`
-	Documentation []BlueprintListContent `json:",omitempty" yaml:",omitempty"`
-	SubBlueprints []BlueprintMiscContent `json:"subBlueprints,omitempty" yaml:"subBlueprints,omitempty"`
-	Examples      []BlueprintMiscContent `json:",omitempty" yaml:",omitempty"`
-}
-
-// BlueprintInterface the input and output variables for the blueprint
-type BlueprintInterface struct {
-	Variables []BlueprintVariable `json:",omitempty" yaml:",omitempty"`
-	// VariableGroups are manually entered
-	VariableGroups []BlueprintVariableGroup `json:"variableGroups,omitempty" yaml:"variableGroups,omitempty"`
-	Outputs        []BlueprintOutput        `json:",omitempty" yaml:",omitempty"`
-}
-
-// BlueprintRequirements defines the roles required and the assocaited services
-// that need to be enabled to provision blueprint resources
-type BlueprintRequirements struct {
-	Roles    []BlueprintRoles
-	Services []string
+	// Title for the solution.
+	// Autogenerated: First H1 text in readme.md
+	Title string
+	// Solution source location and source type
+	// Autogen details in BlueprintRepoDetail
+	Source *BlueprintRepoDetail
+	// Semantic version  for the packaged solution
+	// Autogenerated: From the `module_name` attribute of the `provider_meta "google"` block
+	// E.g.
+	// provider_meta "google" {
+	//  module_name = "blueprints/terraform/terraform-google-log-analysis/v0.1.5"
+	// }
+	Version string `json:",omitempty" yaml:",omitempty"`
+	// Actuation tool e.g. Terraform and its required version
+	// Autogen details in BlueprintActuationTool
+	ActuationTool BlueprintActuationTool `json:"actuationTool,omitempty" yaml:"actuationTool,omitempty"`
+	// Various types of descriptions associated with the solution
+	// Autogen details in BlueprintDescription
+	Description *BlueprintDescription `json:",omitempty" yaml:",omitempty"`
+	// Path to an image representing the icon for the solution
+	// Will be set as "assets/icon.png". Can be manually authored otherwise.
+	Icon string `json:",omitempty" yaml:",omitempty"`
+	// The time estimate for configuring and deploying the solution.
+	// Autogen details in BlueprintTimeEstimate
+	DeploymentTime BlueprintTimeEstimate `json:"deploymentTime,omitempty" yaml:"deploymentTime,omitempty"`
+	// The cost estimate for the solution based on preconfigured variables.
+	// Autogen details in BlueprintCostEstimate
+	CostEstimate BlueprintCostEstimate `json:",omitempty" yaml:",omitempty"`
+	// A list of GCP cloud products used in the solution.
+	// Manually authored.
+	CloudProducts []BlueprintCloudProduct `json:",omitempty" yaml:",omitempty"`
+	// A configuration of fixed and dymanic GCP quotas that apply to the soltuion.
+	// Manually authored.
+	QuotaDetails []BlueprintQuotaDetail `json:",omitempty" yaml:",omitempty"`
+	// Details on the author producing the solution.
+	// Manually authored.
+	Author BlueprintAuthor `json:",omitempty" yaml:",omitempty"`
```

Comment:
nit: Couldn't there be multiple?

Reply:
Based on the requirements, it's only one. If there's multiple people involved, that can specified as part of the Author's description.